integration-tests/src/fees.rs

```diff
@@ -3,35 +3,85 @@
 use crate::kusama_test_net::*;
 use basilisk_runtime::TransactionPayment;
 use frame_support::weights::GetDispatchInfo;
+use primitives::nft::ClassType;
 use sp_runtime::codec::Encode;
 use xcm_emulator::TestExt;
 
+macro_rules! assert_eq_approx {
+	( $x:expr, $y:expr, $z:expr, $r:expr) => {{
+		let diff = if $x >= $y { $x - $y } else { $y - $x };
+		if diff > $z {
+			panic!("\n{} not equal\nleft: {:?}\nright: {:?}\n", $r, $x, $y);
+		}
+	}};
+}
+
+//NOTE: rust encoded call size is differen from UI encoded call size that's why we have asserts for 2 fees.
+
 #[test]
-// #[ignore]
-//TODO: fix this, fees calculcated in this test are apporx 2bsx off.
 fn transfer_transaction_fees() {
-	//This test is not correct
 	Basilisk::execute_with(|| {
+		let diff = UNITS / 100; //0.01
+
+		let expected_rust_encoded_fees = 45_56 * UNITS / 100; //45.56
+		let expected_ui_fees = 4_655 * UNITS / 100; //46.55
+
 		let call = orml_currencies::Call::<basilisk_runtime::Runtime>::transfer {
 			dest: AccountId::from(ALICE),
 			currency_id: 0,
 			amount: 50 * UNITS,
 		};
 
 		let info = call.get_dispatch_info();
+		//rust encoded fees
+		let len = call.encoded_size() as u32;
+		let rust_encoded_fees = TransactionPayment::compute_fee(len, &info, 0);
+
+		assert_eq_approx!(
+			rust_encoded_fees,
+			expected_rust_encoded_fees,
+			diff,
+			"Rust encoded fees are"
+		);
+
+		//UI encoded fees
+		let len = 143; //UI encoded length
```

Comment:
I would call this variable as `ui_encoded_length`, then the comment is not necessary.